### DISCUSSION TITLE
What are the GitHub vs Docker parity expectations?

### DISCUSSION THREAD

**Discussion by jakereps** [openfga/community#290] [General]
Hi 👋 We're currently investigating some issues related to vulnerability scans, and it led me to discover that the docker container for openfga v1.5.1 does not match the source code in Github for v1.5.1. The Dockerfile on v1.5.1 specifies to use the grpc health probe v0.4.25 (fixed version, with no CVEs), but in reality the v1.5.1 openfga image still contains v0.4.24 (triggers CVE scanners).

**Expectation**: 

The v1.5.1 Dockerfile specifies v0.4.25, https://github.com/openfga/openfga/blob/v1.5.1/Dockerfile#L1

```
$ docker run -ti --rm ghcr.io/grpc-ecosystem/grpc-health-probe:v0.4.25@sha256:6cc1dc0af87b35db2ca5fa9b1fbbc389e7570d8ad90ff84a54b6f7ac35cdb423 -addr :8080 -connect-timeout 1h
$ docker cp 81b6492ff918:/ko-app/grpc-health-probe .
Successfully copied 11.8MB to /Users/me/.
$ go version -m grpc-health-probe
grpc-health-probe: go1.22.1
	path	github.com/grpc-ecosystem/grpc-health-probe
	mod	github.com/grpc-ecosystem/grpc-health-probe	(devel)
	dep	github.com/go-jose/go-jose/v3	v3.0.3	h1:fFKWeig/irsp7XD2zBxvnmA/XaRWp5V3CBsZXJF7G7k=
	dep	github.com/golang/protobuf	v1.5.4	h1:i7eJL8qZTpSEXOPTxNKhASYpMn+8e5Q6AdndVa1dWek=
	dep	github.com/spiffe/go-spiffe/v2	v2.1.7	h1:VUkM1yIyg/x8X7u1uXqSRVRCdMdfRIEdFBzpqoeASGk=
	dep	github.com/zeebo/errs	v1.3.0	h1:hmiaKqgYZzcVgRL1Vkc1Mn2914BbzB0IBxs+ebeutGs=
	dep	golang.org/x/crypto	v0.21.0	h1:X31++rzVUdKhX5sWmSOFZxx8UW/ldWx55cbf08iNAMA=
	dep	golang.org/x/net	v0.22.0	h1:9sGLhx7iRIHEiX0oAJ3MRZMUCElJgy7Br1nO+AMN3Tc=
	dep	golang.org/x/sync	v0.6.0	h1:5BMeUDZ7vkXGfEr1x9B4bRcTH4lpkTkpdh0T/J+qjbQ=
	dep	golang.org/x/sys	v0.18.0	h1:DBdB3niSjOA/O0blCZBqDefyWNYveAYMNF1Wum0DYQ4=
	dep	golang.org/x/text	v0.14.0	h1:ScX5w1eTa3QqT8oi6+ziP7dTV1S2+ALU0bI+0zXKWiQ=
	dep	google.golang.org/genproto/googleapis/rpc	v0.0.0-20240308144416-29370a3891b7	h1:em/y72n4XlYRtayY/cVj6pnVzHa//BDA1BdoO+z9mdE=
	dep	google.golang.org/grpc	v1.62.1	h1:B4n+nfKzOICUXMgyrNd19h/I9oH0L1pizfk1d4zSgTk=
	dep	google.golang.org/protobuf	v1.33.0	h1:uNO2rsAINq/JlFpSdYEKIZ0uKD/R9cpdv0T+yoGwGmI=
	build	-buildmode=exe
	build	-compiler=gc
	build	-ldflags="-w -X main.versionTag=${GITHUB_REF_NAME#v}"
	build	-tags=netgo
	build	DefaultGODEBUG=httplaxcontentlength=1,httpmuxgo121=1,panicnil=1,tls10server=1,tlsrsakex=1,tlsunsafeekm=1
	build	CGO_ENABLED=0
	build	GOARCH=amd64
	build	GOOS=linux
	build	GOAMD64=v1
	build	vcs=git
	build	vcs.revision=2c814fcadd44e38afd1755a2b8c036d3576db502
	build	vcs.time=2024-03-10T03:23:19Z
	build	vcs.modified=false
```

`vcs.revision=2c814fcadd44e38afd1755a2b8c036d3576db502` is v0.4.25 of the health probe as expected: https://github.com/search?q=repo%3Agrpc-ecosystem%2Fgrpc-health-probe+2c814fcadd44e38afd1755a2b8c036d3576db502&type=commits

**Reality**
```
 ~ docker ps
CONTAINER ID   IMAGE                    COMMAND          CREATED         STATUS         PORTS     NAMES
b0da865fefad   openfga/openfga:v1.5.1   "/openfga run"   3 seconds ago   Up 3 seconds             quizzical_saha
 ~ docker exec -it b0da865fefad /usr/local/bin/grpc_health_probe --version
${GITHUB_REF_NAME#v}; commit 645566fc4b06a8d51552166a34717ce7090fddf7
 ~ docker cp b0da865fefad:/usr/local/bin/grpc_health_probe .
Successfully copied 11.5MB to /Users/me/.
 ~ go version -m grpc_health_probe
grpc_health_probe: go1.21.5
	path	github.com/grpc-ecosystem/grpc-health-probe
	mod	github.com/grpc-ecosystem/grpc-health-probe	(devel)
	dep	github.com/go-jose/go-jose/v3	v3.0.1	h1:pWmKFVtt+Jl0vBZTIpz/eAKwsm6LkIxDVVbFHKkchhA=
	dep	github.com/golang/protobuf	v1.5.3	h1:KhyjKVUg7Usr/dYsdSqoFveMYd5ko72D+zANwlG1mmg=
	dep	github.com/spiffe/go-spiffe/v2	v2.1.6	h1:4SdizuQieFyL9eNU+SPiCArH4kynzaKOOj0VvM8R7Xo=
	dep	github.com/zeebo/errs	v1.3.0	h1:hmiaKqgYZzcVgRL1Vkc1Mn2914BbzB0IBxs+ebeutGs=
	dep	golang.org/x/crypto	v0.17.0	h1:r8bRNjWL3GshPW3gkd+RpvzWrZAwPS49OmTGZ/uhM4k=
	dep	golang.org/x/net	v0.19.0	h1:zTwKpTd2XuCqf8huc7Fo2iSy+4RHPd10s4KzeTnVr1c=
	dep	golang.org/x/sync	v0.5.0	h1:60k92dhOjHxJkrqnwsfl8KuaHbn/5dl0lUPUklKo3qE=
	dep	golang.org/x/sys	v0.15.0	h1:h48lPFYpsTvQJZF4EKyI4aLHaev3CxivZmv7yZig9pc=
	dep	golang.org/x/text	v0.14.0	h1:ScX5w1eTa3QqT8oi6+ziP7dTV1S2+ALU0bI+0zXKWiQ=
	dep	google.golang.org/genproto/googleapis/rpc	v0.0.0-20231212172506-995d672761c0	h1:/jFB8jK5R3Sq3i/lmeZO0cATSzFfZaJq1J2Euan3XKU=
	dep	google.golang.org/grpc	v1.60.1	h1:26+wFr+cNqSGFcOXcabYC0lUVJVRa2Sb2ortSK7VrEU=
	dep	google.golang.org/protobuf	v1.31.0	h1:g0LDEJHgrBl9N9r17Ru3sqWhkIx2NB67okBHPwC7hs8=
	build	-buildmode=exe
	build	-compiler=gc
	build	-ldflags="-w -X main.versionTag=${GITHUB_REF_NAME#v}"
	build	-tags=netgo
	build	DefaultGODEBUG=panicnil=1
	build	CGO_ENABLED=0
	build	GOARCH=amd64
	build	GOOS=linux
	build	GOAMD64=v1
	build	vcs=git
	build	vcs.revision=645566fc4b06a8d51552166a34717ce7090fddf7
	build	vcs.time=2023-12-18T23:27:55Z
	build	vcs.modified=false
```

`vcs.revision=645566fc4b06a8d51552166a34717ce7090fddf7` is still v0.4.24 of the health probe, and what is included in v1.5.1 of openfga's Docker image: https://github.com/search?q=repo%3Agrpc-ecosystem%2Fgrpc-health-probe+645566fc4b06a8d51552166a34717ce7090fddf7&type=commits

This slightly worries me that there may even be code changes missing. What does the process for release -> docker image updates look like and should something like this be expected, or should they always mirror each other for each stable version cut?

**Comment by rhamzeh** [comment]
Hey @jakereps! Thanks for the finding!

It seems Go Releaser is [configured](https://github.com/openfga/openfga/blob/v1.5.1/.goreleaser.yaml#L29) to use [Dockerfile.goreleaser](https://github.com/openfga/openfga/blob/v1.5.1/Dockerfile.goreleaser) which has not been updated and still uses `0.4.24` of `grpc-health-probe`, we're fixing this in a few and it will be out in the next release.

As for our release process it's entirely [automated](https://github.com/openfga/openfga/blob/main/.github/workflows/release.yaml) once a tag is pushed, including the building of the image and pushing to dockerhub.

We seem to have had some issues with provenance in the `v1.5.1` release (see [action](https://github.com/openfga/openfga/actions/runs/8362142892)), but runs on earlier releases can be found [here](https://github.com/openfga/openfga/actions/workflows/release.yaml), e.g. [v1.5.0](https://github.com/openfga/openfga/actions/runs/8116909894).

Thanks for the heads up! We'll update our dependabot configuration to target that file too.

**Comment by jon-whit** [comment]
https://github.com/openfga/openfga/pull/1503 addresses it.

**Comment by jakereps** [comment]
Ah, that'll do it! Didn't catch the second Dockerfile on a first pass. Thanks all! I suppose I can close this one out 👍

**Comment by jakereps** [comment]
Hey all, this appears to have happened again. I see main has x.26 on `Dockerfile`, and x.25 on `Dockerfile.goreleaser` - we're getting hit with CVE findings against the health probe again 😓

**Reply by jakereps** [reply]
I've opened a PR here bumping the image tag: https://github.com/openfga/openfga/pull/1683

**Comment by jakereps** [comment]
I see you all use Snyk, is there anything that could be done to pre-test images that get pushed to Dockerhub? It seems both images since OpenFGA v1.5.3 would have failed a scan due to the bad health checker. We were previously blocked from upgrading to 1.5.x due to figuring out the authz model reads issue, posted here: https://github.com/openfga/openfga/issues/1668 - so are only now discovering the lingering CVE on the latest version ourselves.
